oshmem/proc/proc.h

```diff
@@ -149,14 +151,50 @@ static inline ompi_proc_t *oshmem_proc_find(int pe)
     return oshmem_proc_for_find(name);
 }
 
+static inline int oshmem_proc_pe_vpid(oshmem_group_t *group, int pe)
+{
+    if (OPAL_LIKELY(pe < group->proc_count)) {
+        return (group->proc_vpids[pe]);
+    } else {
+        return -1;
+    }
+}
+
 static inline int oshmem_proc_pe(ompi_proc_t *proc)
 {
     return (proc ? (int) ((ompi_process_name_t*)&proc->super.proc_name)->vpid : -1);
 }
 
-#define OSHMEM_PROC_JOBID(PROC)    (((ompi_process_name_t*)&((PROC)->super.proc_name))->jobid)
-#define OSHMEM_PROC_VPID(PROC)     (((ompi_process_name_t*)&((PROC)->super.proc_name))->vpid)
-
+#define OSHMEM_PROC_ON_LOCAL_NODE(pe) (opal_bitmap_is_set_bit(&_oshmem_local_vpids, pe))
+static inline int oshmem_proc_init_set_local_vpids()
```

Comment:
Why do we need this functions as inline?
It is not on the data path and doesn't need to be fast.